sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -110,7 +113,7 @@ protected BlobAsyncClientBase(HttpPipeline pipeline, String url, BlobServiceVers
 
         this.accountName = accountName;
         this.containerName = containerName;
-        this.blobName = blobName;
+        this.blobName = Utility.urlEncode(Utility.urlDecode(blobName));
```

Comment:
Don't we encode in the builder, so encoding in the constructor again would be double encoding?

Reply:
See answer about overloads and double encoding above.